coredhcp.go

```diff
@@ -104,11 +104,21 @@ func (s *Server) LoadPlugins(conf *config.Config) ([]*plugins.Plugin, []*plugins
 // It will not reply if the resulting response is `nil`.
 func (s *Server) MainHandler6(conn net.PacketConn, peer net.Addr, req dhcpv6.DHCPv6) {
 	var (
-		resp dhcpv6.DHCPv6
-		stop bool
-		err  error
+		resp, originalReq dhcpv6.DHCPv6
+		stop              bool
+		err               error
 	)
 
+	// decapsulate the relay message
+	if req.IsRelay() {
+		originalReq = req
+		tmp, err := req.GetInnerMessage()
+		if err != nil {
+			log.Warningf("DHCPv6: cannot unpack relay message: %v", err)
```

Comment:
Should it return here instead of continuing?

Reply:
yes, I missed it :(